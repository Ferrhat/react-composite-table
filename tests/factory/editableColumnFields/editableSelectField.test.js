import React from 'react';
import {shallow, mount} from 'enzyme';
import EditableSelectField from '../../../lib/factory/editableColumnFields/editableSelectField';

describe('EditableSelectField', () => {
    const mockOnChange = jest.fn();
    const mockOnClickEditRow = jest.fn();
    const editableSelectField = shallow(<EditableSelectField column={{ name: 'testName', value: 'testValue' }} onClickEditRow={mockOnClickEditRow} rowId={1} onUpdateField={() => Promise.resolve()} />);

    it('renders properly', () => {
        expect(editableSelectField).toMatchSnapshot();
    });

    it('contains a td without Select if the row is not under edit', () => {
        expect(editableSelectField.find('td').length).toEqual(1);
        expect(editableSelectField.find('Select').length).toEqual(0);
    });

    it('contains a td and an Select if the row is under edit', () => {
        editableSelectField.setProps({rowUnderEdit: true, rowUnderEditId: 1, columnUnderEditId: 'testName' }, () => {
            expect(editableSelectField.find('td').length).toEqual(1);
            expect(editableSelectField.find('Select').length).toEqual(1);
        });
    });

});
