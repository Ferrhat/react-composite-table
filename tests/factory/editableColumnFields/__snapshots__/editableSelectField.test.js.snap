// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableSelectField renders properly 1`] = `
<td
  key="testName"
  onClick={[Function]}
/>
`;
